cmd/export.go

```diff
@@ -0,0 +1,86 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package cmd defines command line utilities for ghpc
+package cmd
+
+import (
+	"fmt"
+	"hpc-toolkit/pkg/shell"
+	"log"
+	"os"
+	"path"
+
+	"github.com/spf13/cobra"
+)
+
+func init() {
+	metadataFlag := "metadata"
+	artifactsFlag := "artifacts"
+	exportCmd.Flags().StringVarP(&artifactsDir, artifactsFlag, "a", "", "Alternative artifacts output directory (automatically configured if unset)")
+	exportCmd.Flags().StringVarP(&metadataFile, metadataFlag, "m", "", "Deployment metadata YAML file (automatically configured if unset)")
```

Comment:
What is the best case for having the metadata file as an argument? I am struggling to think of a good case and it just opens the door to more user journeys that we will have to support later. Thoughts on removing it until it is asked for? Also I don't believe it is not in your design doc.

Reply:
I am generally wary of assuming that the directory structure is exactly right else the only thing we can do is fail.

In particular, I believe CI/CD pipelines are likely to exist where the blueprint and the deployment groups "wander off" from one another. Given that the flag is optional and the help indicates "automatically configured" I do not see much pain to supporting it.